space-cli/cmd/code.go

```diff
@@ -106,36 +103,97 @@ func RunDockerFile(s *model.ActionCode, loginResp *model.LoginResponse) error {
-		&container.Config{
-			Image: s.Service.Tasks[0].Docker.Image,
-			Env: []string{
-				"FILE_PATH=/",
-				fmt.Sprintf("URL=%s", s.Service.Tasks[0].Env["URL"]),
-				fmt.Sprintf("TOKEN=%s", loginResp.Token),
-				fmt.Sprintf("meta=%s", string(sa))},
-		},
-		&container.HostConfig{Mounts: []mount.Mount{
-			{
-				Type:   mount.TypeBind,
-				Source: dir,
-				Target: "/build",
-			},
-		}}, nil, "")
+	data, err := ioutil.ReadFile(fmt.Sprintf("%s/Dockerfile", dir))
 	if err != nil {
 		return err
 	}
-	if err := cli.ContainerStart(ctx, resp.ID, types.ContainerStartOptions{}); err != nil {
+	_, err = cli.ImageBuild(ctx, strings.NewReader(string(data)), types.ImageBuildOptions{})
+	if err != nil {
 		return err
 	}
+	// todo store some name or for creating container
 	return nil
 }
+
+func GenerateServiceConfigWithoutLogin() (*model.Service, error) {
+	progLang, err := getProgLang()
+	if err != nil {
+		return nil, err
+	}
+	serviceName := ""
+	if err := survey.AskOne(&survey.Input{Message: "Enter Service Name"}, &serviceName); err != nil {
+		return nil, err
+	}
+	defaultServiceID := strings.ReplaceAll(serviceName, " ", "-")
+	serviceID := ""
+	if err := survey.AskOne(&survey.Input{Message: "Enter Service ID", Default: strings.ToLower(defaultServiceID)}, &serviceID); err != nil {
+		return nil, err
+	}
+	var port int32
+	if err := survey.AskOne(&survey.Input{Message: "Enter Service Port"}, &port); err != nil {
+		return nil, err
+	}
+	projectNameID := ""
+	if err := survey.AskOne(&survey.Input{Message: "Enter Project Name"}, &projectNameID); err != nil {
+		return nil, err
+	}
+
+	dockerImage := ""
+	if err := survey.AskOne(&survey.Input{Message: "Enter Docker Image Name"}, &dockerImage); err != nil {
+		return nil, err
+	}
+	if dockerImage == "" {
+		dockerImage = fmt.Sprintf("%s/%s", projectNameID, serviceID)
+	}
+
+	dockerSecret := ""
+	if err := survey.AskOne(&survey.Input{Message: "Enter Docker Secret"}, &dockerSecret); err != nil {
+		return nil, err
+	}
+
+	secret := ""
+	if err := survey.AskOne(&survey.Input{Message: "Enter File & Environment Secret (CSV)"}, &secret); err != nil {
+		return nil, err
+	}
+	secrets := []string{}
+	for _, value := range strings.Split(secret, ",") {
+		secrets = append(secrets, value)
+	}
+
+	var progCmd string
+	if err := survey.AskOne(&survey.Input{Message: "Enter Run Cmd", Default: strings.Join(getCmd(progLang), " ")}, &progCmd); err != nil {
+		return nil, err
+	}
+
+	c := &model.Service{
+		ID:        serviceID,
+		Name:      serviceName,
+		ProjectID: projectNameID,
+		Version:   "v1",
+		Scale:     model.ScaleConfig{Replicas: 0, MinReplicas: 0, MaxReplicas: 100, Concurrency: 50},
+		Tasks: []model.Task{
+			{
+				ID:        serviceID,
+				Name:      serviceName,
+				Ports:     []model.Port{{Name: "http", Protocol: "http", Port: port}},
+				Resources: model.Resources{CPU: 250, Memory: 512},
+				Docker:    model.Docker{Image: dockerImage, Secret: dockerSecret},
```

Comment:
Set the secret only if its not none. A simple way would be to set the `dockerSecret` value to `""` if its value was none